cram/thread_pool.c

```diff
@@ -434,17 +449,33 @@ t_pool *t_pool_init(int qsize, int tsize) {
     p->total_time = p->wait_time = 0;
 #endif
 
-    p->t = malloc(tsize * sizeof(p->t[0]));
+    if (!(p->t = malloc(tsize * sizeof(p->t[0]))))
+        goto fail;
 
-    pthread_mutex_init(&p->pool_m, NULL);
-    pthread_cond_init(&p->empty_c, NULL);
-    pthread_cond_init(&p->full_c, NULL);
+    if (0 != pthread_mutex_init(&p->pool_m, NULL)) goto fail;
```

Comment:
For the record, Yoda conditionals are not the convention in extant samtools/htslib source code, and probably some of the maintainers hate them.  And they're kind of pointless when the operator is `!=`…